cluster-autoscaler/core/scale_down.go

```diff
@@ -755,14 +736,24 @@ func (sd *ScaleDown) SoftTaintUnneededNodes(allNodes []*apiv1.Node) (errors []er
 
 // TryToScaleDown tries to scale down the cluster. It returns a result inside a ScaleDownStatus indicating if any node was
 // removed and error if such occurred.
-func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, pdbs []*policyv1.PodDisruptionBudget,
-	currentTime time.Time) (*status.ScaleDownStatus, errors.AutoscalerError) {
+func (sd *ScaleDown) TryToScaleDown(pdbs []*policyv1.PodDisruptionBudget, currentTime time.Time) (*status.ScaleDownStatus, errors.AutoscalerError) {
+
 	scaleDownStatus := &status.ScaleDownStatus{NodeDeleteResults: sd.nodeDeletionTracker.GetAndClearNodeDeleteResults()}
 	nodeDeletionDuration := time.Duration(0)
 	findNodesToRemoveDuration := time.Duration(0)
 	defer updateScaleDownMetrics(time.Now(), &findNodesToRemoveDuration, &nodeDeletionDuration)
-	nodesWithoutMaster := filterOutMasters(allNodes, pods)
-	candidates := make([]*apiv1.Node, 0)
+
+	allNodeInfos, errSnapshot := sd.context.ClusterSnapshot.NodeInfos().List()
+	if errSnapshot != nil {
+		// This should never happen, List() returns err only because scheduler interface requires it.
```

Comment:
I do not believe this comment is helpful.

Reply:
We've discussed it with @MaciekPytel before and it seems to be the least evil version for now. Ideally I'd prefer to have an error-less method to use internally, since none of the snapshot implementations can return error here:s

Reply:
Fair. I was just suggesting to remove the comment. But this is nit.